File: code/jvm/im-http-api/src/main/kotlin/im/api/model/output/PaginationOutputModel.kt
package im.api.model.output

import im.repository.pagination.PaginationInfo

data class PaginationOutputModel(
    val total: Long?,
    val totalPages: Int?,
    val current: Int,
    val next: Int?,
    val previous: Int?,
) {
    companion object {
        fun fromPagination(pagination: PaginationInfo?): PaginationOutputModel? {
            if (pagination == null) {
                return null
            }
            return PaginationOutputModel(
                total = pagination.total,
                totalPages = pagination.totalPages,
                current = pagination.currentPage,
                next = pagination.nextPage,
                previous = pagination.prevPage,
            )
        }
    }
}

File: code/client/web/src/Dto/output/PaginationOutputModel.ts
/**
 * Output model for pagination information.
 *
 * @param total The total number of items.
 * @param totalPages The total number of pages.
 * @param current The current page number.
 * @param next The next page number.
 * @param previous The previous page number.
 */
export interface PaginationOutputModel {
    total: number | null;
    totalPages: number | null;
    current: number;
    next: number | null;
    previous: number | null;
}
